x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java

```diff
@@ -470,7 +491,34 @@ void searchVersionedDocumentForSubject(Subject subject, ActionListener<Versioned
         });
     }
 
-    private void createNewProfile(Subject subject, String uid, ActionListener<Profile> listener) throws IOException {
+    private static final String INVALID_USERNAME_MESSAGE = "Security domain [%s] is configured to use literal username. "
+        + "As a result, creating new user profile requires the username to be at least 1 and no more than 490 characters. "
+        + "The username can contain alphanumeric characters (a-z, A-Z, 0-9), spaces, punctuation, "
+        + "and printable symbols in the Basic Latin (ASCII) block.";
+
+    private ValidationException validateUsername(Subject subject) {
```

Comment:
I think this validation format (to return an exception) is ugly and should be confined to request body validation. I would prefer we just throw exceptions.

Reply:
I changed to throw `ElasticsearchException`